### ISSUE TITLE
Terminal Emulator that explains Pasted Commands In Simple Terms to Newcomers

### ISSUE THREAD

**Issue by mamamia88** [open-source-ideas/open-source-ideas#270]
## Project description
People considering switching to linux are often turned away by guides that mention the command line but, once you get used too it it's often a good way to accomplish certain tasks quicker than you could otherwise. The cli shouldn't be something to be feared it is a great tool designed by humans for humans and I came up with a relatively fun and easy way to train people which is a terminal application that either splits horizontally or vertically that references the man pages and internet databases to ELI5 to newcomers after which they are given a y/n do you want to run this prompt. 

It could even have some form of community chat built in where the user asks the community for help like if it's safe to run the commands etc prior to running the commands. Maybe the community could be used to sort out bad advice online and push it further down in the search results or be removed by the moderators of distro specific forums take notice and either delete the search result from google or correct it. 

Relevant Tech: Quite honestly I'm a moron and not really sure I'll try to edit this as recommendations come in or not.

**Comment by KOLANICH** [comment]
I remember there was an app, that executed another app within a kind of extremily poor sandbox implemented using LD_PRELOAD with functions redefining libc functions and then printing the summary of FS actions done by that app.

But I cannot google it right now.

An another idea how it can look like is an app parsing `--help` message into a kind of a spec for a CLI parser like in `plumbum` for Python and `CLI11` for C++, then parsing the CLI using this parser, and then instead of doing actions printing the help on the attrs used. I feel like it may make sense to implement it in CLI11 and plumbum themselves.

**Comment by KaKi87** [comment]
Similar but non-CLI project : https://github.com/idank/explainshell

**Comment by KaKi87** [comment]
Instead of creating a terminal emulator, I would like to suggest creating a simple `explain` command, like there's one on SQL, that you put before the command, and which will output the answer.

**Comment by FredrikAugust** [comment]
Please resubmit this with the format specified.

**Comment by FredrikAugust** [comment]
Or if you wish to edit this issue to comply with the guidelines I can reopen the issue.

**Comment by KaKi87** [comment]
Hey, the shell command actually exists :)
https://benjamine.github.io/explain.sh/
